app/data_model/answer_store.py

```diff
@@ -6,25 +6,27 @@
 
 
 class Answer:
-    def __init__(self, answer_id, value, group_instance=0, answer_instance=0):
+    def __init__(self, answer_id, value, group_instance_id=None, group_instance=0, answer_instance=0):
         if answer_id is None or value is None:
             raise ValueError("Both 'answer_id' and 'value' must be set for Answer")
 
         self.answer_id = answer_id
+        self.group_instance_id = group_instance_id
         self.group_instance = group_instance
         self.answer_instance = answer_instance
         self.value = value
 
     def matches(self, answer):
```

Comment:
This is more of a discussion point: Should this still use `group_instance` on the matching, because surely the `group_instance_id` is the only one that matters now?

Reply:
This PR already updates form helper to fetch answers by `(answer_id,group_instance_id)` so I believe it should be safe to do the same when saving.  I'll have a more thorough read through the code to be sure.